### ISSUE TITLE
Not returning loss in GPTLMHeadModel implementation

### ISSUE THREAD

**Issue by danxunLi** [Dao-AILab/flash-attention#419]
Hi Professor Tri Dao,

I might be completely missing something, but I saw that the GPTLMHeadModel implementation [here](https://github.com/Dao-AILab/flash-attention/blob/main/flash_attn/models/gpt.py#L459) does not return a loss calculation in the forward call. (In fact it looks like only the BERT implementation does.) 

The forward call also does not accept any other features except for input_ids. I am using the HuggingFace Trainer, which expects the loss in the output from the model if the training data only has input_ids (no labels, no attention mask). I feel like I am missing something huge? Should I be implementing the loss calculation in the GPTLMHeadModel forward call above?

Guidance is appreciated.

Thanks so much!

**Comment by tridao** [comment]
Right, we don't return the loss, unlike Huggingface's implementation. I apply a loss function outside this module. This allows more flexibility in modeling and experimentation (e.g. changing the loss function).
You can write your own wrapper to wrap GPTLMHeadModel that takes in input_ids & labels and return the loss.
